### PR TITLE
[C#] feat: O1 model support

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #2103 

## Details
Added support for `o1-preview` and `o1-mini` models.

* Bumped `OpenAI` and `Azure.AI.OpenAI` to `2.1.0-beta.1`.
* Tested o1 support with light bot sample with monologue augmentation.
* Updated `teamsChefBot-streaming` to use deployed `Microsoft.Teams.AI` nuget package.
* Fixed `LayoutSection` bug with incorrect ordering of sections.

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by corinagum — CHANGES_REQUESTED** [review]
Do we want to merge this into the C# dev branch instead of main?

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/OpenAIModel.cs [hunk 1]
```diff
@@ -185,7 +191,11 @@ public async Task<PromptResponse> CompletePromptAsync(ITurnContext turnContext,
                 };
             }
 
-            if (!_options.UseSystemMessages!.Value && prompt.Output.Count > 0 && prompt.Output[0].Role == ChatRole.System)
+            // Get the model to use.
+            string model = promptTemplate.Configuration.Completion.Model ?? _deploymentName;
+            bool isO1Model = model.StartsWith("o1-");
```
Comment (lilyydu) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/OpenAIModel.cs:196]: do we know how streaming works with the o1 model?

File: dotnet/samples/04.ai.g.teamsChefBot-streaming/TeamsChefBot.csproj [hunk 2]
```diff
@@ -16,18 +16,7 @@
     <PackageReference Include="Microsoft.Bot.Connector" Version="4.22.9" />
     <PackageReference Include="Microsoft.KernelMemory.AI.OpenAI" Version="0.73.240906.1" />
     <PackageReference Include="Microsoft.KernelMemory.Core" Version="0.73.240906.1" />
-	  <!-- PackageReference Include="Microsoft.Teams.AI" Version="1.5.*" /-->
```
Comment (lilyydu) [inline dotnet/samples/04.ai.g.teamsChefBot-streaming/TeamsChefBot.csproj:19]: Thank you for catching this!

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/OpenAIModel.cs (hunk 1 above)
Comment (singhk97) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/OpenAIModel.cs:196]: https://openai.com/index/introducing-openai-o1-preview/#:~:text=include%20function%20calling%2C-,streaming,-%2C%20support%20for%20system - nope it does not support that.

**Comment by singhk97** [comment]
> Do we want to merge this into the C# dev branch instead of main?

Good catch - updated.

**Comment by singhk97** [comment]
I have tested the streaming samples - and it works.

**Review by corinagum — APPROVED** [review]
LGTM!

**Review by lilyydu — APPROVED** [review]

**Pull request merged**